lib/datadog/tracing.rb

```diff
@@ -0,0 +1,118 @@
+      # This means it is not advisable to cache it.
+      #
+      # TODO: I think this next paragraph can be better written.
+      # Most of the functionality available through the {.tracer} instance is
+      # also available in public methods in the {Datadog::Tracing} module.
+      # It is preferable to use the public methods in the {Datadog::Tracing} when possible
+      # as {Datadog::Tracing} strongly defines the tracing public API, and thus
+      # we strive to no introduce breaking changes to {Datadog::Tracing} methods.
+      #
+      # @return [Datadog::Tracer] the active tracer
+      def tracer
+        components.tracer
+      end
+
+      # The active, unfinished trace, representing the current instrumentation context.
+      #
+      # The active trace is thread-local.
+      #
+      # @return [Datadog::TraceSegment] the active trace
+      # @return [nil] if no trace is active
+      def active_trace
+        tracer.active_trace
+      end
+
+      # The active, unfinished span, representing the currently instrumented application section.
+      #
+      # The active span belongs to an {.active_trace}.
+      #
+      # @see .active_trace
+      #
+      # @return [Datadog::SpanOperation] the active span
+      # @return [nil] if no trace is active, and thus no span is active
+      def active_span
+        tracer.active_span
+      end
+
+      # If an active trace is present, forces it to be retained by the Datadog backend.
+      #
+      # Any sampling logic will not be able to change this decision.
+      #
+      # @return [void]
+      def keep!
+        active_trace.keep!
+      end
+
+      # If an active trace is present, forces it to be dropped and not stored by the Datadog backend.
+      #
+      # TODO: should we mention billing? do we know if this directly affects billing?
+      #
+      # Any sampling logic will not be able to change this decision.
+      #
+      # @return [void]
+      def reject!
+        active_trace.reject!
+      end
+
+      # Information about the currently active trace that allows
+      # for another execution context to be linked to the active
+      # trace.
+      #
+      # The most common use case is for propagating distributed
+      # tracing information to downstream services.
+      #
+      # @return [Datadog::Correlation::Identifier] correlation object
+      def correlation
+        tracer.active_correlation
+      end
+
+      # Textual representation of {.correlation}, which can be
+      # added to individual log lines in order to correlate them with the active
+      # trace.
+      #
+      # Example:
+      #
+      # ```
+      # MyLogger.log("#{Datadog::Tracing.log_correlation}] My log message")
+      # # dd.env=prod dd.service=billing dd.version=13.8 dd.trace_id=545847825299552251 dd.span_id=711755234730770098 My log message
+      # ```
+      #
+      # @return [String] correlation information
+      def log_correlation
+        correlation.to_s
+      end
```

Comment:
Minor: This may get a bit expensive -- if you emit a few hundred log lines for a single span, we create the correlation object that many times just to generate the same `to_s`.

Reply:
Added a comment about memoization to `Datadog::Correlation::Identifier`.